drivers/flash/soc_flash_nrf.c

```diff
@@ -94,32 +94,24 @@ static inline bool is_aligned_32(uint32_t data)
 	return (data & 0x3) ? false : true;
 }
 
-static inline bool is_regular_addr_valid(off_t addr, size_t len)
+static inline bool is_within_bounds(off_t addr, size_t len, off_t boundary_start,
+				    size_t boundary_size)
 {
-	size_t flash_size = nrfx_nvmc_flash_size_get();
-
-	if (addr >= flash_size ||
-	    addr < 0 ||
-	    len > flash_size ||
-	    (addr) + len > flash_size) {
-		return false;
-	}
-
-	return true;
+	return !(addr >= boundary_start + boundary_size ||
```

Comment:
If that got already moved, maybe change it to:
```
return (addr >= boundary_start &&
        (addr < (boundary_start + boundary_size)) &&
        (len <= (boundary_start + boundary_size - addr)));
```

this should also cover wrapping of `boundary_start + boundary_size` and `addr + len`; and as long as we can ensure `boundary_start >= 0` the `addr < 0 ` is also covered.

Reply:
I'm busy with the upmerge. Can someone check the boolean logic here ?

And also that there does not exist any possible evil input on addr or len that could cause a security issue.

boundary_start and boundary_size can be trusted.

Reply:
by

> this should also cover wrapping of `boundary_start + boundary_size` (...)

I meant the lines that I have provided; I was not suggesting the existing solution does not do that, I was arguing that proposed is (I have not botched it).

Reply:
Oh, well, IMHO we don't need to do input-checking on the boundary args, as we can inspect every possible input.

But I guess it does make the code future-proof.

I agree with the change, I just would like someone else to review.

e.g. what happens if addr or len is negative?

Reply:
Do not spend on this more time.

Reply:
applied the change